In [50]:
# Save the url on variable 'src'

src = "https://raw.githubusercontent.com/Biuni/PokemonGO-Pokedex/master/pokedex.json"

In [51]:
# Import pandas for loading data and data manipulation

import pandas as pd

In [52]:
# Read the data as a pandas DataFrame and check the first 5 rows

df = pd.read_json(src)
df.head()

pokemon
0  {'id': 1, 'num': '001', 'name': 'Bulbasaur', '...
1  {'id': 2, 'num': '002', 'name': 'Ivysaur', 'im...
2  {'id': 3, 'num': '003', 'name': 'Venusaur', 'i...
3  {'id': 4, 'num': '004', 'name': 'Charmander', ...
4  {'id': 5, 'num': '005', 'name': 'Charmeleon', ...

In [54]:
# Check the shape of the dataset

df.shape

(151, 1)

We have 151 records. But all the rows are as dictionaries. We need unpack them.

In [55]:
# Unpacking the dataset as a list

s = df.stack()
s_list = [*s]

In [57]:
# Creating a new dataframe with the unpacked dictionaries

df_copy = pd.DataFrame(s_list)

df_copy.head()

id  num        name                                               img  \
0   1  001   Bulbasaur  http://www.serebii.net/pokemongo/pokemon/001.png   
1   2  002     Ivysaur  http://www.serebii.net/pokemongo/pokemon/002.png   
2   3  003    Venusaur  http://www.serebii.net/pokemongo/pokemon/003.png   
3   4  004  Charmander  http://www.serebii.net/pokemongo/pokemon/004.png   
4   5  005  Charmeleon  http://www.serebii.net/pokemongo/pokemon/005.png   

              type  height    weight             candy  candy_count  \
0  [Grass, Poison]  0.71 m    6.9 kg   Bulbasaur Candy         25.0   
1  [Grass, Poison]  0.99 m   13.0 kg   Bulbasaur Candy        100.0   
2  [Grass, Poison]  2.01 m  100.0 kg   Bulbasaur Candy          NaN   
3           [Fire]  0.61 m    8.5 kg  Charmander Candy         25.0   
4           [Fire]  1.09 m   19.0 kg  Charmander Candy        100.0   

           egg  spawn_chance  avg_spawns spawn_time multipliers  \
0         2 km         0.690        69.0      20:00      [1.58]   
1  Not in Eggs         0.042         4.2      07:00  [1.2, 1.6]   
2  Not in Eggs         0.017         1.7      11:30        None   
3         2 km         0.253        25.3      08:45      [1.65]   
4  Not in Eggs         0.012         1.2      19:00      [1.79]   

                     weaknesses  \
0  [Fire, Ice, Flying, Psychic]   
1  [Fire, Ice, Flying, Psychic]   
2  [Fire, Ice, Flying, Psychic]   
3         [Water, Ground, Rock]   
4         [Water, Ground, Rock]   

                                      next_evolution  \
0  [{'num': '002', 'name': 'Ivysaur'}, {'num': '0...   
1               [{'num': '003', 'name': 'Venusaur'}]   
2                                                NaN   
3  [{'num': '005', 'name': 'Charmeleon'}, {'num':...   
4              [{'num': '006', 'name': 'Charizard'}]   

                                      prev_evolution  
0                                                NaN  
1              [{'num': '001', 'name': 'Bulbasaur'}]  
2  [{'num': '001', 'name': 'Bulbasaur'}, {'num': ...  
3                                                NaN  
4             [{'num': '004', 'name': 'Charmander'}]

Data Attributes:
- id: Identification Number - int 
- num: Number of the Pokémon in the official Pokédex - int 
- name: Pokémon name - string 
- img: URL to an image of this Pokémon - string 
- type: Pokémon type - string 
- height: Pokémon height - float
- weight: Pokémon weight - float 
- candy: type of candy used to evolve Pokémon or given when transferred - string 
- candy_count: the amount of candies required to evolve - int
- egg: Number of kilometers to travel to hatch the egg - float 
- spawn_chance: Percentage of spawn chance (NEW) - float 
- avg_spawns: Number of this pokemon on 10.000 spawns (NEW) - int
- spawn_time: Spawns most active at the time on this field. Spawn times are the same for all time zones and are expressed in local time. (NEW) - “minutes: seconds” 
- multipliers: Multiplier of Combat Power (CP) for calculating the CP after evolution See below - list of int
- weakness: Types of Pokémon this Pokémon is weak to - list of strings 
- next_evolution: Number and Name of successive evolutions of Pokémon - list of dict 
- prev_evolution: Number and Name of previous evolutions of Pokémon - list of dict

In [58]:
# Removing the 'km' in eggs column so that numerals can be read as they are

df_copy["egg"] = df_copy["egg"].apply(lambda x: x.replace(" km", "") if "km" in x else x)

In [59]:
# Removing the ' m' in height column and converting to float

df_copy["height"] = df_copy.height.str.replace(" m", "").astype(float)

In [60]:
# Removing the ' kg' in weight column and converting to float

df_copy["weight"] = df_copy.weight.str.replace(" kg", "").astype(float)

In [61]:
# Checking the dataset datatypes

df_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 151 entries, 0 to 150
Data columns (total 17 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              151 non-null    int64  
 1   num             151 non-null    object 
 2   name            151 non-null    object 
 3   img             151 non-null    object 
 4   type            151 non-null    object 
 5   height          151 non-null    float64
 6   weight          151 non-null    float64
 7   candy           151 non-null    object 
 8   candy_count     70 non-null     float64
 9   egg             151 non-null    object 
 10  spawn_chance    151 non-null    float64
 11  avg_spawns      151 non-null    float64
 12  spawn_time      151 non-null    object 
 13  multipliers     70 non-null     object 
 14  weaknesses      151 non-null    object 
 15  next_evolution  70 non-null     object 
 16  prev_evolution  72 non-null     object 
dtypes: float64(5), int64(1), object(11)

In [62]:
# Export DataFrame to excel file

df_copy.to_excel("pokemon.xlsx", index=False, header=True)